### ISSUE TITLE
[suggestion] turn off service from notification 

### ISSUE THREAD

**Issue by uomar8** [StefanIlchev/ElementumService#6]
it would be good if you could turn off the service from notification

 because sometimes i close Kodi from recent apps and this this doesn't turn off the service so i have to open Kodi again open Elementum again and reclose it from the app

**Comment by StefanIlchev** [comment]
[Foreground services](https://developer.android.com/guide/components/foreground-services) have notifications, [that's just how they are](https://developer.android.com/guide/components/foreground-services#start). I chose to use a foreground service instead of a background service because the system should be less eager to stop it when it needs resources as doing so will have a noticeable effect on the user's experience which it will - if the service stops so will Elementum.

I don't know how to distinguish you closing Kodi from recent apps and Kodi being stopped by the system for being in the background for too long. I sometimes like to navigate away from Kodi and do something else while I leave it to buffer up and don't want the buffering to stop just because Kodi was stopped.

In conclusion, if you want to completely shut Kodi down and leave nothing running, then you're going to have to do it from the app.

**Comment by ChristianCiach** [comment]
I think @uomar8 is asking for a way to manually stop the elementum service from the android tv home screen, via a notification.

Yes, android tv also has notifications that you can interact with, just like regular Android. But I don't know if they are created using the same API.

**Comment by StefanIlchev** [comment]
@ChristianCiach that's not a bad idea. I'll look into showing the notification on TV and adding a close button to it when I have the time. Thanks!

**Comment by StefanIlchev** [comment]
It's done, clicking the notification now stops the service. However, it seems like only system apps can directly show notifications on TV and there doesn't seem to be a command that solves this. But it looks like a TV might show a notification when you have a media session, so I added a dummy media session.

It seems to be working as intended on Google TV - you access the notification by holding the "Home" button and it stays there while the service is running.

On Android TV on the other hand the notification can be dismissed and disappears when you open another app. Also, a "Now Playing" card is shown in the launcher with two buttons - "Open" and "Stop". The former doesn't seem to do anything and the latter pauses the media session instead of stopping it and dismisses the card. I couldn't find much information on this card so I left it like that.

@uomar8, could you please try and see if the latest version works for you. Also, which version of Android are you using and is it a Google TV or an Android TV?

**Comment by uomar8** [comment]
Android 10 works 
Android 8.1 TV box works as well 

For me the problem solved

**Comment by ChristianCiach** [comment]
I can test this later today on my nvidia shield TV.

**Comment by StefanIlchev** [comment]
@uomar8, @ChristianCiach could you please try the version I just released when you have the time. I had to correct two really dumb things that I missed.

**Comment by ChristianCiach** [comment]
Seems a bit wonky to me. 

* The notification says "Unknown - Close app". I don't know where the "Unknown" comes from.
* Sometimes when I properly exit kody (using "Exit" from the power menu) the notification persists.
* Sometimes, even though I know that the service is still running (Torrents are still downloading and seeding in Elementum), the notification is missing.

I think it's really nice to have a notification to see if the service is still running, but it only seems to work like 75% of the time for me.

Tested with 0.1.86.19

**Comment by StefanIlchev** [comment]
@ChristianCiach thanks for the feedback. Could you please try and see if 0.1.86.20 improves things?

* The notification uses "Unknown" when the metadata it wants is missing and it can't be bothered to fallback to something else. I added more metadata which should hopefully resolve the problem. If not, I can add more. It has to be one of [these](https://developer.android.com/reference/android/media/MediaMetadata).
* This is probably because I never explicitly set the media session to inactive and just destroyed it assuming that it'll imply inactivity. I corrected that in the new version.
* I noticed something similar on Android TV. When I play something from another app its media session notification will replace mine but when I stop it my notification won't come back. Google TV doesn't have this problem and restores my media session notification. Also, the notification isn't dismissible on Google TV.
